source/base/quadrature.cc

```diff
@@ -297,17 +367,26 @@ QAnisotropic<dim>::QAnisotropic(const Quadrature<1> &qx)
       this->weights[k++] = qx.weight(k1);
     }
   Assert (k==this->size(), ExcInternalError());
+  this->is_tensor_product_flag = true;
 }
 
 
 
 template <int dim>
 QAnisotropic<dim>::QAnisotropic(const Quadrature<1> &qx,
                                 const Quadrature<1> &qy)
-  : Quadrature<dim>(qx.size()
-                    *qy.size())
+  :  Quadrature<dim>(qx.size()*qy.size())
 {
   Assert (dim==2, ExcImpossibleInDim(dim));
+}
+
+
+
+template <>
+QAnisotropic<2>::QAnisotropic(const Quadrature<1> &qx,
+                              const Quadrature<1> &qy)
+  :  Quadrature<2>(qx.size()*qy.size())
```

Comment:
Ah, I see now -- this is where we call the constructor with just an integer. Ignore my comment from above then.